thirdeye-ui/src/app/components/subscription-group-view/alert-associations-view-table/alert-associations-view-table.utils.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 StarTree Inc
+ *
+ * Licensed under the StarTree Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { UiSubscriptionGroupAlert } from "../../../rest/dto/ui-subscription-group.interfaces";
+import {
+    getAssociationId,
+    getEnumerationItemName,
+} from "../../subscription-group-wizard/alerts-dimensions/alerts-dimensions.utils";
+import { UiAssociation } from "./alert-associations-view-table.interface";
+
+export const getUiAssociation = (
+    uiSubscriptionGroupAlerts: UiSubscriptionGroupAlert[],
+    t: (v: string, args?: Record<string, string>) => string
+): UiAssociation[] => {
+    const uiAssociations: UiAssociation[] = [];
+
+    (uiSubscriptionGroupAlerts || []).forEach((alert) => {
```

Comment:
Let's leave it to the parent caller to do the null/undefined check for `uiSubscriptionGroupAlerts` when passing the parameter and assume `uiSubscriptionGroupAlerts` is an array here

Reply:
Moved the null check out of the function implementation, to parent function caller.

In the cases where there may or may not be null data, I tend to add a null check even if ensuring the validity of the data is someone else's responsibility, to ensure that the user at-least gets a running page with no data, instead of a page crash.

Reply:
If `uiSubscriptionGroupAlerts ` is still null, that would mean the user that used this function is not doing the checks appropriately

Reply:
Got it. So the responsibility should always be with the implementer, and failing that should indicate an issue in the implementation (instead of adding fallback values everywhere)